pkg/security/podsecuritypolicy/seccomp/mustrunas.go

```diff
@@ -57,67 +57,70 @@ func (s *withSeccompProfile) Validate(pod *api.Pod, container *api.Container) fi
 
 	if len(s.allowedProfiles) == 0 {
 		// if anything is set it is invalid.
-		return s.validateNoAnnotationsAreSet(pod, annotationsPath)
+		return s.validateNoProfilesAreSet(pod, annotationsPath)
 	}
 
 	// at least pod should have something set and any container annotation
 	// must also be valid
-	return s.validateAnnotationsAreAllowed(pod, annotationsPath)
+	return s.validateProfilesAreAllowed(pod, annotationsPath)
 }
 
-func (s *withSeccompProfile) validateAnnotationsAreAllowed(pod *api.Pod, fieldPath *field.Path) field.ErrorList {
+// validateProfilesAreAllowed validates that any set profile falls within the allowable profiles
+// of the strategy.
+func (s *withSeccompProfile) validateProfilesAreAllowed(pod *api.Pod, fieldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
+	podSpecFieldPath := field.NewPath("pod", "spec")
 
-	podProfile, hasPodProfile := pod.Annotations[api.SeccompPodAnnotationKey]
 	// empty pod profile is allowed IF all the containers have their own valid annotation.
 	// If the container's do not have their own annotations we'll catch it in the logic
 	// below.
-	if hasPodProfile && !isProfileAllowed(podProfile, s.allowedProfiles) {
-		allErrs = append(allErrs, field.NotSupported(fieldPath.Child(api.SeccompPodAnnotationKey), podProfile, s.allowedProfiles))
+	podProfile := ""
+	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.SeccompProfile != nil {
+		podProfile = *pod.Spec.SecurityContext.SeccompProfile
+		if !isProfileAllowed(*pod.Spec.SecurityContext.SeccompProfile, s.allowedProfiles){
+			allErrs = append(allErrs, field.NotSupported(podSpecFieldPath.Child("securityContext", "seccompProfile"),
```

Comment:
I changed the error to `Forbidden` for when you aren't allowed to set anything.  I feel like `NotSupported` is more appropriate here since it is validating against a set of enumerated values.  Here are the descriptions for the two, let me know if you still feel strongly about it:

```
// Forbidden returns a *Error indicating "forbidden".  This is used to
// report valid (as per formatting rules) values which would be accepted under
// some conditions, but which are not permitted by current conditions (e.g.
// security policy).
```

```
// NotSupported returns a *Error indicating "unsupported value".
// This is used to report unknown values for enumerated fields (e.g. a list of
// valid values).
```